From: Ibrahim ARI <ibrahim.ari@pardus.org.tr>
Date: Thu, 11 Nov 2021 12:03:36 +0300
Subject: update reporsitory path

---
 main.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main.go b/main.go
index 330a8e6..ba42124 100644
--- a/main.go
+++ b/main.go
@@ -5,8 +5,8 @@ import (
 	"log"
 	_ "net/http/pprof"
 	"runtime"
-
-	"github.com/nim4/DBShield/dbshield"
+	"vcs.ahtapot.org.tr/ahtapot/dbshield/dbshield"
+//	"github.com/nim4/DBShield/dbshield"
 )
 
 func usage(showUsage bool) {
